homeassistant/components/dsmr_reader/sensor.py

```diff
@@ -1,30 +1,48 @@
 """Support for DSMR Reader through MQTT."""
+import logging
+
 from homeassistant.components import mqtt
 from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE
 from homeassistant.core import callback
 from homeassistant.util import slugify
 
 from .definitions import SENSORS, DSMRReaderSensorEntityDescription
 
+_LOGGER = logging.getLogger(__name__)
+
 DOMAIN = "dsmr_reader"
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up DSMR Reader sensors."""
-    async_add_entities(DSMRSensor(description) for description in SENSORS)
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
```

Comment:
Those are not related to this integration and thus cannot be used for a unique ID in this integration.

For other options see: <https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements>

Reply:
Ok, I will try to use the serial number which is part of the raw mqtt telegram. 
Downside is that you need to wait for a message to setup the sensors.

Reply:
You can use the config entry + description key as last resort option

Reply:
Ok but than I have to add a config flow right?

Reply:
true